desktop-widgets/filterwidget2.cpp

```diff
@@ -114,8 +114,8 @@ FilterWidget2::FilterWidget2(QWidget* parent) :
 		this, &FilterWidget2::updatePlanned);
 
 	// Update temperature fields if user changes temperature-units in preferences.
-	connect(qPrefUnits::instance(), &qPrefUnits::temperatureChanged,
-		this, &FilterWidget2::temperatureChanged);
+	connect(qPrefUnits::instance(), SIGNAL(temperatureChanged(int)),
+		this, SLOT(temperatureChanged()));
```

Comment:
switching back to the old syntax seems like the wrong solution.

Reply:
hmmm according to my knowledge using the macros is actually the new syntax, but I might be wrong. 

Connect without parameters should only be used when connecting signal to signal.

Reply:
I'm not the Qt expert here, but I am 99% sure that it used to be just the macros, which cause type checking to happen at run time and then (I think maybe with Qt5) the function address syntax was added.
Yeah, I think the wiki confirms that : https://wiki.qt.io/New_Signal_Slot_Syntax

Reply:
you are correct, and I strongly believe type checking is helping to secure our source against address violations.

But if you prefer, I will remove the added check.

Reply:
I think I am misunderstanding you, or you me.
Or I am misunderstanding the change in Qt's syntax.
I was under the impression that the new syntax did a better job in type checking at compile time.
But this isn't something that I am comfortable enough arguing about, so I'll accept the change.

Reply:
look at the different statement, if you write the type of parameters the compiler can surely control it better :-)

Reply:
See my comments on #2452 - the Qt documentation appears to flatly disagree with you on this one.
https://doc.qt.io/qt-5/signalsandslots-syntaxes.html
So let's not do this.